src/jwt/signatureVerifier.js

```diff
@@ -44,21 +45,30 @@ export const verifySignature = (idToken, options) => {
     return Promise.resolve(payload);
   }
 
-  return getJwk(options.domain, header.kid).then(jwk => {
-    const pubKey = KEYUTIL.getKey(jwk);
-    const signatureValid = KJUR.jws.JWS.verify(idToken, pubKey, ['RS256']);
+  return getJwk(options.domain, header.kid)
+    .then(jwk => rsaVerifierForKey(jwk))
+    .then(rsaVerifier => {
+      const encodedParts = idToken.split('.');
+      const headerAndPayload = encodedParts[0] + '.' + encodedParts[1];
+      const signature = base64.decodeToHEX(encodedParts[2]);
 
-    if (signatureValid) {
-      return Promise.resolve(payload);
-    }
+      if (rsaVerifier.verify(headerAndPayload, signature)) {
+        return Promise.resolve(payload);
+      }
 
-    return Promise.reject(
-      idTokenError({
-        error: 'invalid_signature',
-        desc: 'Invalid token signature',
-      }),
-    );
-  });
+      return Promise.reject(
+        idTokenError({
+          error: 'invalid_signature',
+          desc: 'Invalid ID token signature',
+        }),
+      );
+    });
+};
+
+const rsaVerifierForKey = jwk => {
+  const modulus = base64.decodeToHEX(jwk.n);
+  const exponent = base64.decodeToHEX(jwk.e);
+  return Promise.resolve(new RSAVerifier(modulus, exponent));
```

Comment:
Why make a Promise for this non-async operation?

Reply:
Will refactor ⚡️

Reply:
I think this was like this to be able to reject the promise if there was a "runtime error" on the verifier instantiation. We will need to check.